nest/04.finest.md

```diff
@@ -0,0 +1,191 @@
+It's going to add a new requirement to the graph, which will be fulfilled by, for example, `stable::sys-lib/glibc#1.8.1`.
+
+```
+@root
+├─ stable::shell/dash#*
+|   └─ fulfilled by stable::shell/dash#0.5.10
+|      └─ stable::sys-lib/glibc#>=1
+|         └─ fulfilled by stable::sys-lib/glibc#1.8.1
+|            └─ and so on, until there are no depedencies left
+└─ ...
+```
+
+Now, if you want to create a group of packages, the graph would then look like so:
+
+```
+@root
+├─ stable::shell/dash#*
+|   └─ fulfilled by stable::shell/dash#0.5.10
+|      └─ stable::sys-lib/glibc#>=2
+|         └─ fulfilled by stable::sys-lib/glibc#2.0.1
+|            └─ and so on, until there are no depedencies left
+├─ @Group1
+|  ├─ requirement1
+|  |  └─ fullfiled by package1  
+|  └─ requirement2
+|     └─ fulfilled by package2
+└─ ...
+```
+
+## 4.2 finest commands
+### 4.2.1 Managing packages through requirements
+#### Add a requirement
+The command `finest requirement add <requirement> [--parent <group>]` adds a requirement to a group.
+By default, this group is `@root`.
+With the `--parent` flag, you can speficy a parent group.
+
+When adding a requirement, `finest` makes a copy of your current dependency graph (if a copy wasn't already available), adds that requirement to its parent and fulfills the dependencies of the package recursively.
+
+Its output looks like this:
+
+```
+$ finest requirement add linux
+Adding requirement stable::kernel/linux#* to group @root...
+```
+
+#### Remove a requirement
+`finest requirement remove <requirement> [--parent <group>]` removes a requirement from a group.
+The group is, by default, `@root`, but you can specify a parent group with the flag `--parent`.
+
+When removing a requirement, `finest` makes a copy of your current dependency graph (if a copy wasn't already made), removes that requirement from its parent and removes its depedencies recursively if possible.
+
+Its output is similar to this:
+
+```
+$ finest requirement remove linux --parent @work
+Removing requirement stable::kernel/linux#* to group @work...
+```
+
+#### Common issues
+* `error: no package found for requirement <req>`
+
+This error means that `finest` couldn't find any package to satisfy this requirement.
+What probably went wrong, is that the syntax of the requirement is incorrect.
+You can check on [Raven-OS website](https://stable.raven-os.org/) to check the repository and the category of this requirement.
+
+### 4.2.2 Updating the system by merging the depedency graph
+#### Merging the dependency graph
+`finest merge` merges the copy of your dependency graph, on which `finest` applied the transactions, to your current dependency graph.
+These two graphs are compared to extract a list of transactions that `finest` will perform, in order to go from one graph to another.
+And so, your dependency graph will finally apply the transactions you meant to do.
```

Comment:
I don't think the dependency graph is the one who does that, rather finest right?
```suggestion
And then, `finest` will finally apply the transactions you meant to do.
```